azext_iot/central/providers/device_provider.py

```diff
@@ -181,8 +189,12 @@ def get_device_registration_info(
         if info:
             return info
 
-        device = self.get_device(device_id)
-        if device.device_status == DeviceStatus.provisioned:
+        device_essential_info = self.get_device_essential_info(device_id)
```

Comment:
My suggestion: just do `device = self.get_device(device_id)` and then in the if statement below do:

```
if device.status == DeviceStatus.provisioned:
  ...
  info = {
    ..
    "device_info": device.get_device_registration_info()
  }
```